cmd/tke-installer/app/installer/expansion.go

```diff
@@ -0,0 +1,206 @@
+/*
+ * Tencent is pleased to support the open source community by making TKEStack
+ * available.
+ *
+ * Copyright (C) 2012-2019 Tencent. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the “License”); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an “AS IS” BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package installer
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"tkestack.io/tke/cmd/tke-installer/app/installer/constants"
+)
+
+// expansionPath is the dir where users put there expansion files. default is data/expansions/
+var expansionPath string
+
+// expansionEnabled
+var expansionEnabled bool
+
+const envExpansionPath = "EXPANSION_PATH"
+const expansionDirName = "expansions/"
+const defaultExpansionPath = constants.DataDir + expansionDirName
+
+// expansion layout
+const expansionLayoutChartsDir = "charts/"
+const expansionLayoutFilesDir = "files/"
+const expansionLayoutFilesGeneratedDir = "files_generated/"
+const expansionLayoutHooksDir = "hooks/"
+const expansionLayoutProviderDir = "provider/"
+const expansionLayoutImagesName = "images.tar.gz" //nolint
+const expansionLayoutConfDir = "conf/"
+const expansionLayoutApplicationsDir = "applications/"
+
+var expansionLayoutMap map[string]string
+
+// expansion resource
+const expansionResourceChart = "chart"
+const expansionResourceFile = "file"
+const expansionResourceFileGenerated = "file_generated"
+const expansionResourceHook = "hook"
+const expansionResourceProvider = "provider"
+const expansionResourceConf = "conf"
+const expansionResourceApplication = "application"
+
+// expansion value
+const expansionValuesFile = "values.yaml" //nolint
+
+func init() {
+
+	expansionPath = os.Getenv(envExpansionPath)
```

Comment:
getting from env is not a good way, getting from user input, like json config, is better.